docs/model_server_c_api.md

```diff
@@ -0,0 +1,45 @@
+* There is support for stateful models.
+
+## API Description
+
+Server functionalities are encapsulated in shared library built from OVMS source. To include OVMS you need to link this library with your application and use C API defined in [header file](../src/ovms.h). To run OVMS you need to spawn process that will keep OVMS alive. Then you can schedule inference both directly from app using C API and gRPC/HTTP endpoints.
+
+### Server configuration
+
+To start OVMS you need to create `OVMS_Server` object using `OVMS_ServerNew`, with set of `OVMS_ServerSettings` and `OVMS_ModelsSettings` that describe how the server should be configured. Once the server is started using `OVMS_ServerStartFromConfigurationFile` you can schedule the inferences using `OVMS_Inference`. Example how to use OVMS with C/C++ application is [here](../demos/c_api_minimal_app/README.md).
+
+### Error handling
+Most of OVMS C API functions return `OVMS_Status` object pointer indicating the success or failure. Success is indicated by nullptr. Failure is indicated by returning `OVMS_Status` object. The status code can be extracted using `OVMS_StatusGetCode` function and the details of error can be retrieved using `OVMS_StatusGetDetails` function.
+
+The ownership of `OVMS_Status` is passed to the caller of the function. You must delete the object using `OVMS_StatusDelete`.
+
+### Inference
+
+To execute inference using C API you must follow steps described below.
+
+#### Prepare inference request
+Create an inference request using `OVMS_InferenceRequestNew` specifying which servable name and optionally version to use. Then specify input tensors with `OVMS_InferenceRequestAddInput` and set the tensor data using `OVMS_InferenceRequestSetData`.
+
+#### Invoke inference
+Execute inference with OVMS using `OVMS_Inference` synchronous call. During inference execution you must not modify `OVMS_InferenceRequest` and bound memory buffers.
+
+#### Process inference response
+If the inference was successful, you receive `OVMS_InferenceRequest` object. After processing the response, you must free with `OVMS_InferenceResponseDelete`.
+
+To process response, you first must check for inference error. If no error occurred, you must iterate over response outputs and parameters using `OVMS_InferenceResponseGetOutputCount` and `OVMS_InferenceResponseGetParameterCount`. Then you must extract details about each output and parameter using `OVMS_InferenceResponseGetOutput` and `OVMS_InferenceResponseGetParameter`. Example how to use OVMS with C/C++ application is [here](../demos/c_api_minimal_app/README.md). While in example app you have only single thread scheduling inference request you can execute multiple inferences simultaneously using different threads.
+
+**Note**: After inference execution is finished you can reuse the same `OVMS_InferenceRequest` by using `OVMS_InferenceRequestInputRemoveData` and then setting different tensor data with `OVMS_InferenceRequestSetData`.
```

Comment:
Can you omit `OVMS_InferenceRequestInputRemoveData` and just overwrite with `OVMS_InferenceRequestSetData`? That would be easier for an user

Reply:
At this moment you can't.